rust/connlib/tunnel/src/client.rs

```diff
@@ -183,13 +226,17 @@ where
             return Ok(());
         };
 
-        if let Some(dns_packet) = dns::parse(&tunnel.resources.read(), packet.as_immutable()) {
-            if let Err(e) = send_dns_packet(&device_writer, dns_packet) {
-                tracing::error!(err = %e, "failed to send DNS packet");
-                let _ = tunnel.callbacks.on_error(&e.into());
+        match dns::parse(&tunnel.resources.read(), packet.as_immutable()) {
+            Some(dns::ResolveStrategy::LocalResponse(pkt)) => {
+                if let Err(e) = send_dns_packet(&device_writer, pkt) {
+                    tracing::error!(err = %e, "failed to send DNS packet");
+                    let _ = tunnel.callbacks.on_error(&e.into());
+                }
```

Comment:
`continue` needs to be here because we've identified and handled the packet as a DNS packet, right?

Reply:
Right! on any of the matches we want to do that.